src/bitcoin-cli.cpp

```diff
@@ -928,6 +938,46 @@ static void GetProgressBar(double progress, std::string& progress_bar)
     }
 }
 
+/**
+ * GetFeerateBar constructs fee rate distribution bars using dynamic increment based on size.
+ *
+ * @param[in]  size      Size of fee rate group in bytes.
+ * @param[out] size_bar  String representation of the distribution bar.
+ * @param[in]  inc_size  Minimum size of fee rate groups in bytes.
+ * @param[in]  max_size  Maximum size of fee rate groups in bytes.
+ */
+static void GetFeerateBar(double size, std::string& size_bar, double inc_size, double max_size)
+{
+    if (size == 0) return;
+
+    static double INCREMENT{inc_size};
+    static const std::string BAR{"\u2591"};
+
+    if (max_size < 10000){
+        for (int i = 0; i < size / INCREMENT; ++i) {
+            size_bar += BAR;
+        }
+    } else if (max_size > 10000 && max_size < 100000){
+        for (int i = 0; i < size / (INCREMENT*10); ++i) {
+            size_bar += BAR;
+        }
+    } else if (max_size > 100000 && max_size < 1000000){
+    	for (int i = 0; i < size / (INCREMENT*100); ++i) {
+            size_bar += BAR;
+        }
```

Comment:
All the for-cycles are the same basically, so maybe you can have one for-cycle after this set of conditions. That may actually be easier to review.

Reply:
Right this can be done. Will do later today.